packages/react-query/src/shared/hooks/types.ts

```diff
@@ -139,9 +139,22 @@ export interface UseTRPCMutationOptions<
 export interface UseTRPCSubscriptionOptions<TOutput, TError> {
   enabled?: boolean;
   onStarted?: () => void;
+  onConnecting?: () => void;
   onData: (data: TOutput) => void;
   onError?: (err: TError) => void;
 }
+
+export type SubscriptionState = 'connecting' | 'connected' | 'stopped';
```

Comment:
- idle
- pending
- error

Reply:
Maybe connecting as well, as a pre-state of pending

Reply:
Idle means subscription is not enabled?